src/Microsoft.AspNet.Mvc.Core/ParameterBinding/ModelBindingHelper.cs

```diff
@@ -532,5 +532,17 @@ public static object ConvertValuesToCollectionType<T>(Type modelType, IList<T> v
                 return null;
             }
         }
+
+        private static Type[] GetTypeArgumentsIfMatch([NotNull] Type closedType, Type matchingOpenType)
```

Comment:
This has always been a strange method.  Why not use `TypeExtensions.ExtractGenericInterface()`?  Current implementation is much more constrained.